pkg/controller/llmisvc/workload_single_node.go

```diff
@@ -272,3 +300,133 @@ func semanticDeploymentIsEqual(expected *appsv1.Deployment, curr *appsv1.Deploym
 		equality.Semantic.DeepDerivative(expected.Labels, curr.Labels) &&
 		equality.Semantic.DeepDerivative(expected.Annotations, curr.Annotations)
 }
+
+func hasRoutingSidecar(pod corev1.PodSpec) bool {
+	return routingSidecar(&pod) != nil
+}
+
+func routingSidecar(pod *corev1.PodSpec) *corev1.Container {
+	for i := range pod.InitContainers {
+		if pod.InitContainers[i].Name == routingSidecarContainerName {
+			return &pod.InitContainers[i]
+		}
+	}
+	return nil
+}
```

Comment:
I am not sure this belongs here. Especially that it's used across different topology reconciles. Maybe we can have `workload_[checks|detectors].go`  ... I am terrible at naming but this is better than `_utils.go`.

Reply:
sure, it's small enough, I missed it. Lets move those funcs + const there.